Missing go module sigs.k8s.io/kind

### Description

The required go module `sigs.k8s.io/kind` is missing:

```
no required module provides package sigs.k8s.io/kind; to add it:
	go get sigs.k8s.io/kind
make[1]: *** [/Users/gerbermichi/dev/control-api/local-env/../local-env/kind-kubeconfig-v1.22.1] Error 1
```

### Additional Context

_No response_

### Logs

```shell
no required module provides package sigs.k8s.io/kind; to add it:
	go get sigs.k8s.io/kind
make[1]: *** [/Users/gerbermichi/dev/control-api/local-env/../local-env/kind-kubeconfig-v1.22.1] Error 1
```

### Expected Behavior

I should not have to install the module manually.

### Steps To Reproduce

1. `make local-env-setup`
2. Accept default URL
3. Enter name `mig`
4. Import realm into keycloak
5. Create admin user in keacloak
6. Press the enter key in the terminal


### Version used

v1.22.1

### Version of OpenShift

default

--- a/.gitignore
+++ b/.gitignore
@@ -31,6 +31,7 @@ crd*.yaml
 /local-env/realm.json
 /local-env/.created
 /local-env/.kind-setup_complete
+/local-env/webhook-certs
 
 # Go releaser
 dist/

--- a/Makefile
+++ b/Makefile
@@ -64,11 +64,11 @@ clean: ## Cleans up the generated resources
 
 .PHONY: run-api
 KUBECONFIG ?= ~/.kube/config
-run-api: build ## Starts control api against the configured kuberentes cluster
+run-api: build ## Starts control api apiserver against the current Kubernetes cluster (based on your local config)
 	$(BIN_FILENAME) api --secure-port 9443 --kubeconfig $(KUBECONFIG) --authentication-kubeconfig $(KUBECONFIG) --authorization-kubeconfig $(KUBECONFIG) --cluster-roles appuio-organization-viewer,appuio-organization-admin --username-prefix "appuio#"
 
 .PHONY: run-controller
-run-controller: build ## Starts control api against the configured kuberentes cluster
+run-controller: build ## Starts control api controller against the current Kubernetes cluster (based  on your local config)
 	$(BIN_FILENAME) controller --username-prefix "appuio#"
 
 .PHONY: local-env

--- a/go.mod
+++ b/go.mod
@@ -6,6 +6,7 @@ require (
 	github.com/golang/mock v1.6.0
 	github.com/spf13/cobra v1.4.0
 	github.com/stretchr/testify v1.7.1
+	go.uber.org/multierr v1.8.0
 	k8s.io/api v0.23.5
 	k8s.io/apimachinery v0.23.5
 	k8s.io/apiserver v0.23.5
@@ -13,6 +14,7 @@ require (
 	sigs.k8s.io/apiserver-runtime v1.0.3-0.20210913073608-0663f60bfee2
 	sigs.k8s.io/controller-runtime v0.11.1
 	sigs.k8s.io/controller-tools v0.8.0
+	sigs.k8s.io/kind v0.12.0
 	sigs.k8s.io/kustomize/kustomize/v4 v4.5.3
 )
 
@@ -28,6 +30,7 @@ require (
 	github.com/NYTimes/gziphandler v1.1.1 // indirect
 	github.com/PuerkitoBio/purell v1.1.1 // indirect
 	github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578 // indirect
+	github.com/alessio/shellescape v1.4.1 // indirect
 	github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a // indirect
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/blang/semver v3.5.1+incompatible // indirect
@@ -37,6 +40,7 @@ require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/emicklei/go-restful v2.9.5+incompatible // indirect
 	github.com/evanphx/json-patch v4.12.0+incompatible // indirect
+	github.com/evanphx/json-patch/v5 v5.6.0 // indirect
 	github.com/fatih/color v1.12.0 // indirect
 	github.com/felixge/httpsnoop v1.0.1 // indirect
 	github.com/form3tech-oss/jwt-go v3.2.3+incompatible // indirect
@@ -73,6 +77,7 @@ require (
 	github.com/monochromegane/go-gitignore v0.0.0-20200626010858-205db1a8cc00 // indirect
 	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 // indirect
 	github.com/olekukonko/tablewriter v0.0.5 // indirect
+	github.com/pelletier/go-toml v1.9.4 // indirect
 	github.com/pkg/errors v0.9.1 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/prometheus/client_golang v1.11.0 // indirect
@@ -97,7 +102,6 @@ require (
 	go.opentelemetry.io/proto/otlp v0.7.0 // indirect
 	go.starlark.net v0.0.0-20200306205701-8dd3e2ee1dd5 // indirect
 	go.uber.org/atomic v1.7.0 // indirect
-	go.uber.org/multierr v1.8.0 // indirect
 	go.uber.org/zap v1.21.0 // indirect
 	golang.org/x/crypto v0.0.0-20210921155107-089bfa567519 // indirect
 	golang.org/x/mod v0.4.2 // indirect

--- a/local-env/kind.mk
+++ b/local-env/kind.mk
@@ -18,8 +18,12 @@ kind-clean: ## Remove the kind Cluster
 ### Artifacts
 ###
 
+webhook-certs/tls.key:
+	mkdir -p webhook-certs
+	openssl req -x509 -newkey rsa:4096 -nodes -keyout webhook-certs/tls.key -out webhook-certs/tls.crt -days 3650 -subj "/CN=webhook-service.control-api.svc" -addext "subjectAltName = DNS:webhook-service.control-api.svc"
+
 $(KIND_KUBECONFIG): export KUBECONFIG = $(KIND_KUBECONFIG)
-$(KIND_KUBECONFIG):
+$(KIND_KUBECONFIG): webhook-certs/tls.key
 	$(localenv_dir)/setup-kind.sh "$(KIND)" "$(KIND_CLUSTER)" "$(KIND_NODE_VERSION)" "$(KIND_KUBECONFIG)"
 	@kubectl version
 	@kubectl cluster-info

--- a/local-env/setup-kind.sh
+++ b/local-env/setup-kind.sh
@@ -90,6 +90,52 @@ kubectl config set-context --current --user=oidc-user
 kubectl apply -k "${script_dir}/../config/crd/apiextensions.k8s.io/v1"
 kubectl apply -k "${script_dir}/../config/deployment"
 kubectl apply -k "${script_dir}/../config/user-rbac"
+kubectl apply -k "${script_dir}/../config/webhook"
+kubectl create secret tls -n control-api webhook-service-tls --cert=webhook-certs/tls.crt --key=webhook-certs/tls.key
+kubectl -n control-api patch deployment control-api-controller \
+  --type=json \
+  -p '[
+    {
+      "op": "add",
+      "path": "/spec/template/spec/containers/0/args/-",
+      "value": "--webhook-cert-dir=/var/run/webhook-service-tls"
+    },
+    {
+      "op": "add",
+      "path": "/spec/template/spec/volumes",
+      value: [
+        {
+          "name": "webhook-service-tls",
+          "secret": {
+            "secretName": "webhook-service-tls"
+          }
+        }
+      ]
+    },
+    {
+      "op": "add",
+      "path": "/spec/template/spec/containers/0/volumeMounts",
+      "value": [
+        {
+          "name": "webhook-service-tls",
+          "mountPath": "/var/run/webhook-service-tls",
+          "readOnly": true
+        }
+      ]
+    }
+  ]'
+kubectl patch validatingwebhookconfiguration validating-webhook-configuration \
+  -p '{
+    "webhooks": [
+      {
+        "name": "validate-users.appuio.io",
+        "clientConfig": {
+          "caBundle": "'"$(base64 -w0 "${script_dir}"/webhook-certs/tls.crt)"'"
+        }
+      }
+    ]
+  }'
+
 
 echo =======
 echo "Setup finished. To interact with the local dev cluster, set the KUBECONFIG environment variable as follows:"

--- a/local-env/setup-kind.sh
+++ b/local-env/setup-kind.sh
@@ -38,11 +38,13 @@ fi
 echo
 identifier=
 while [ x"$identifier" == x"" ]; do
-  read -r -p "Provide an identifier for your local-dev Keycloak realm: " identifier
+  read -r -p "Provide a suffix for your local-dev Keycloak realm (all local-dev realms are prefixed with 'local-dev-'): " identifier
 done
 
 realm_name="local-dev-${identifier}"
 sed -e "s/REPLACEME/${realm_name}/g" "${script_dir}/templates/realm.json.tpl" > "${script_dir}/realm.json"
+
+echo -e "\033[1mUsing '${realm_name}' as your local-dev Keycloak realm\033[0m"
 
 step "Navigate to ${keycloak_url} and create a new realm by importing the '$(realpath "${script_dir}/realm.json")' file."
 
@@ -91,5 +93,5 @@ kubectl apply -k "${script_dir}/../config/user-rbac"
 
 echo =======
 echo "Setup finished. To interact with the local dev cluster, set the KUBECONFIG environment variable as follows:"
-echo "\"export \$KUBECONFIG=$(realpath "${kind_kubeconfig}")\""
+echo "\"export KUBECONFIG=$(realpath "${kind_kubeconfig}")\""
 echo =======

--- a/tools.go
+++ b/tools.go
@@ -12,4 +12,6 @@ import (
 	_ "sigs.k8s.io/kustomize/kustomize/v4"
 	// To generate mocks
 	_ "github.com/golang/mock/mockgen"
+	// To have Kind updated via Renovate
+	_ "sigs.k8s.io/kind"
 )

